### PR TITLE
Initial unit tests

### PR DESCRIPTION
**Pull request by stevenaleung**
- Created initial unit tests using ideal sinusoids
- Addressed issue #8

To do:
- Define I_spta
- Make metrics robust to DC offset / additive noise
- Use noisy data for testing

### CODE REVIEW

File: src/models.jl [hunk 1]
```diff
@@ -33,14 +33,16 @@ struct Excitation
     pulse_duration: On-time / duration of the excitation pulse in seconds.
```
Comment (mofii) [inline src/models.jl:33]: The documentation needs to be updated here as well.

File: test/runtests.jl [hunk 2]
```diff
@@ -3,22 +3,34 @@ using TestSetExtensions
...
+    # waveform parameters
+    dt = 1/sampling_freq_receive
+    num_samples = ceil(pulse_duration/dt)
+    time = collect((0:num_samples-1)*dt)
+    data = sin.(2*pi*center_freq*time)*pressure_amplitude
+    
+    # intensity tests
+    intensity_expected = uconvert(u"W/m^2", pressure_amplitude^2 / (M.density*M.c))
+    @test intensity(pressure_amplitude, M) ≈ intensity_expected atol=0.001u"W/m^2"
+    @test intensity(uconvert(u"Pa", pressure_amplitude), M) ≈ intensity_expected atol=0.001u"W/m^2"
```
Comment (mofii) [inline test/runtests.jl:26]: Nice that you got approx to work with units!

File: src/models.jl [hunk 3]
```diff
@@ -33,14 +33,16 @@ struct Excitation
...
 struct Excitation
-    f0::herz                # Center frequency in Hz
-    pulse_duration::seconds # seconds
-    duty_cycle::Float64     # [0,1]
-    total_duration::seconds
+    f0::hertz                   # Center frequency in Hz
+    pulse_duration::seconds     # seconds
+    duty_cycle::Float64         # [0,1]
+    total_duration::seconds     # seconds
+    sampling_frequency_receive::hertz
```
[the remark this answers is not in the document]
Comment (stevenaleung) [reply src/models.jl:44]: Sounds good I'll make the change

File: test/runtests.jl [hunk 4]
```diff
@@ -3,22 +3,34 @@ using TestSetExtensions
...
+    @test intensity(pressure_amplitude, M) ≈ intensity_expected atol=0.001u"W/m^2"
+    @test intensity(uconvert(u"Pa", pressure_amplitude), M) ≈ intensity_expected atol=0.001u"W/m^2"   
 
-        I = map(x->intensity(x, M), data)
-        sppa = intensity_sppa(data, M, E)
-        spta = intensity_spta(data, M, E)
-    end
+    # intensity_sppa tests. output is expected to be a vector
+    intensity_sppa_expected = uconvert(u"W/cm^2", pressure_amplitude^2 / (2*M.density*M.c))
+    @test intensity_sppa(data, M, E) ≈ [intensity_sppa_expected] atol=0.001u"W/cm^2"
```
Comment (stevenaleung) [inline test/runtests.jl:30]: Intensity_sppa() currently outputs a 1 element vector. Do we want it to output a single Quantity value instead?

File: src/models.jl (hunk 1 above)
Comment (stevenaleung) [reply src/models.jl:33]: On it!

**Comment by stevenaleung** [comment]
> * What did you mean by defining an I_spta() function?

We discussed briefly this morning - which part of the waveform should be used to calculate I_spta. Look forward to reading the document/schematic you were referring to

**Comment by mofii** [comment]
What's the current status of this PR?
I see you we expecting a document/schematic from me. I have no recollection of that. Did you ever get it?

File: test/runtests.jl (hunk 4 above)
Comment (stevenaleung) [reply test/runtests.jl:30]: Vector output should be kept so that broadcasting is available

**Comment by stevenaleung** [comment]
Didn't get the document/schematic, but the calculation of I_spta looks ok to me

**Review by mofii — APPROVED** [review]

**Pull request merged**